shell/src/main/java/alluxio/cli/fsadmin/decommission/StartDecommissionCommand.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.fsadmin.decommission;
+
+import alluxio.cli.CommandUtils;
+import alluxio.cli.fsadmin.FileSystemAdminShellUtils;
+import alluxio.cli.fsadmin.command.AbstractFsAdminCommand;
+import alluxio.cli.fsadmin.command.Context;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.exception.status.AlluxioStatusException;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Start worker decommission process.
+ */
+public class StartDecommissionCommand extends AbstractFsAdminCommand {
+  private static final Logger LOG = LoggerFactory.getLogger(StartDecommissionCommand.class);
+
+  private static final Option HELP_OPTION =
+      Option.builder("h")
+          .required(false)
+          .hasArg(false)
+          .desc("print help information.")
+          .build();
+  private static final Option EXCLUDED_HOSTS_OPTION =
+      Option.builder()
+          .longOpt("excluded-hosts")
+          .required(false)
+          .hasArg(true)
+          .numberOfArgs(1)
+          .argName("excluded-hosts")
+          .desc("A list of excluded worker hosts separated by comma.")
+          .build();
+  private static final Option EXCLUDED_HOSTS_INPUT_FILE =
+      Option.builder()
+          .longOpt("input-file")
+          .required(false)
+          .hasArg(true)
+          .numberOfArgs(1)
+          .argName("input-file")
+          .desc("The file specific workers to be decommissioned.")
+          .build();
+  private static final Option EXCLUDED_HOSTS_CLEAR =
+      Option.builder("clear")
+          .required(false)
+          .hasArg(false)
+          .desc("If true, clear the old decommissioned map from master, and update with entries "
+              + "collected this time, so the workers are not chosen this time will release "
+              + "from the decommissioned map, and they have a chance to reactive."
```

Comment:
I don't understand what is this for? A decommissioned worker cannot "recover" from the decommissioned state unless it is restarted. Clearing the decommissioned worker set does not automatically restart the worker and make it re-register again.

Reply:
Same as above, the clear option give us a chance to reactive workers that not mentioned in this time. But using a shell command to do this is better choice.